executor/src/witgen/machines/block_machine.rs

```diff
@@ -588,7 +570,13 @@ impl<'a, T: FieldElement> WitnessColumnEvaluator<T> for WitnessData<'a, T> {
         };
         // It is not an error to access the rows outside the block.
         // We just return a symbolic ID for those.
-        match self.data[&id].get(row as usize).cloned().flatten() {
+
+        let column = self
+            .data
+            .get(&id)
+            .ok_or(crate::witgen::IncompleteCause::NotConcrete)?;
```

Comment:
If the `get` fails, this means we are evaluating a polynomial constraint that contains a reference to a witness column this block machine is not prepared to handle. But the machine extractor creates the machines just in the way that it contains all witness columns that are connected via polynomial constraints. So I think this should be a hard assertion here.
Can you explain which example triggers this and on what column (and what do the logs say about "found a machine with columns ..."?)

Reply:
Ok, I think this is caused by the fact that we have one block machine constructed for `main.latch, main.x` and another one for `main.latch, main.y`, but to identify the block machine, we only use the selector. So I think the proper fix would be to change BlockMachine::process_plookup to not only check that `right.selector != self.selector`, but also that all witness columns in `right` are keys in `self.data`.

Reply:
I mean of course a better way to do this would be to somehow store in the main machine which sub machine is responsible to handle which lookup instead of looping through all machines in the hope that one of them volunteers.

Reply:
Note that there can be one machine responsible for multiple lookups, mostly if the RHS of the lookups is the same.

Reply:
Right, so the main thing in this PR is the test case. I can even rebase to only keep that as I think it should pass but it doesn't.